### PR TITLE
Sync with upstream march-to-cpu-opt

### PR DESCRIPTION
**Pull request by patrick-rivos**
This sets mask undisturbed data to all 1's
This potentially has perf implications.

Copy/paste of upstream with zicond/zvbb added to qemu_ext_opts.

### CODE REVIEW

**Review by ewlu — APPROVED** [review]

**Review by ewlu — APPROVED** [review]
Approved with good staging run

File: scripts/march-to-cpu-opt [hunk 1]
```diff
@@ -121,31 +149,61 @@ def get_vlen(ext_dict):
...
-        cpu_opt.append("d=false")
-    return ",".join(cpu_opt)
+        cpu_options.append("f=false")
+        cpu_options.append("d=false")
+
+    return ",".join(cpu_options)
+
+def print_spike_isa():
+    cpu_options = []
+    cpu_options.append("rv{0}".format(CPU_OPTIONS['xlen']))
```
Comment (kevinl-rivosinc) [inline scripts/march-to-cpu-opt:189]: If the string is initialized here, f"rv{CPU_OPTIONS['xlen']}" might be easier to read

File: scripts/march-to-cpu-opt [hunk 2]
```diff
@@ -121,31 +149,61 @@ def get_vlen(ext_dict):
...
+
+    for ext in CPU_OPTIONS['extensions']:
+        if ext not in SPIKE_EXT_NOT_ALLOWED:
+            normalized_ext_name = ext
+
+            if len(ext) > 1:
+                normalized_ext_name = "_{0}".format(normalized_ext_name)
```
Comment (kevinl-rivosinc) [inline scripts/march-to-cpu-opt:196]: ditto

File: scripts/march-to-cpu-opt [hunk 3]
```diff
@@ -121,31 +149,61 @@ def get_vlen(ext_dict):
...
+
+            cpu_options.append(normalized_ext_name)
+
+    return "".join(cpu_options)
 
+def print_spike_varch():
+    if not CPU_OPTIONS['vlen']:
+        return ""
+
+    return "vlen:{0},elen:{1}".format(CPU_OPTIONS['vlen'], CPU_OPTIONS['elen'])
```
Comment (kevinl-rivosinc) [inline scripts/march-to-cpu-opt:206]: ditto

**Review by kevinl-rivosinc — COMMENTED** [review]
else lgtm

File: scripts/march-to-cpu-opt (hunk 1 above)
Comment (patrick-rivos) [reply scripts/march-to-cpu-opt:189]: Discussed and while I agree it'd be better to fix upstream and pull it in.

**Pull request merged**